include/host/wasi/environ.h

```diff
@@ -38,6 +40,73 @@ class Environ {
 
   void fini() noexcept;
 
+  WasiExpect<void> freeAddrInfo(__wasi_addrinfo_t *Addrinfo) {
+    free(Addrinfo);
+    return {};
+  }
+  __wasi_addrinfo_t *addrinfoHelper(__wasi_addrinfo_t *DstAddrinfo,
+                                    addrinfo *SrcAddrinfo) {
+    if (SrcAddrinfo == nullptr) {
+      return nullptr;
+    }
+    DstAddrinfo->ai_flags =
+        static_cast<__wasi_aiflags_t>(SrcAddrinfo->ai_flags);
+    DstAddrinfo->ai_family =
+        static_cast<__wasi_address_family_t>(SrcAddrinfo->ai_family);
+    DstAddrinfo->ai_socktype =
+        static_cast<__wasi_sock_type_t>(SrcAddrinfo->ai_socktype);
+    DstAddrinfo->ai_protocol =
+        static_cast<__wasi_protocol_t>(SrcAddrinfo->ai_protocol);
+    DstAddrinfo->ai_addrlen = SrcAddrinfo->ai_addrlen;
+
+    auto TmpAddress = new __wasi_address_t();
+    memcpy(reinterpret_cast<void *>(TmpAddress->buf), SrcAddrinfo->ai_addr,
+           SrcAddrinfo->ai_addrlen);
+    TmpAddress->buf_len = SrcAddrinfo->ai_addrlen;
+    memcpy(reinterpret_cast<void *>(DstAddrinfo->ai_addr), TmpAddress,
+           sizeof(__wasi_address_t));
+    free(TmpAddress);
+
+    memcpy(reinterpret_cast<void *>(DstAddrinfo->ai_canonname),
+           SrcAddrinfo->ai_canonname, strlen(SrcAddrinfo->ai_canonname));
+    DstAddrinfo->ai_canonname_len = strlen(SrcAddrinfo->ai_canonname);
+
+    DstAddrinfo->ai_next = reinterpret_cast<uint64_t>(addrinfoHelper(
+        reinterpret_cast<__wasi_addrinfo_t *>(DstAddrinfo->ai_next),
+        SrcAddrinfo->ai_next));
+    memcpy(reinterpret_cast<void *>(DstAddrinfo->ai_next),
+           addrinfoHelper(
+               reinterpret_cast<__wasi_addrinfo_t *>(DstAddrinfo->ai_next),
+               SrcAddrinfo->ai_next),
+           sizeof(__wasi_addrinfo_t));
+
+    return DstAddrinfo;
+  }
+
+  WasiExpect<void> getAddrInfo(const char *Node, const char *Service,
+                               const __wasi_addrinfo_t *Hint,
+                               /*Out*/ __wasi_addrinfo_t **Res) {
+    struct addrinfo TmpHint;
+    struct addrinfo *TmpResult = nullptr;
+
+    TmpHint.ai_flags = Hint->ai_flags;
+    TmpHint.ai_family = Hint->ai_family;
+    TmpHint.ai_socktype = Hint->ai_socktype;
+    TmpHint.ai_protocol = Hint->ai_protocol;
+    TmpHint.ai_addrlen = Hint->ai_addrlen;
+    TmpHint.ai_addr = reinterpret_cast<sockaddr *>(
+        reinterpret_cast<__wasi_address_t *>(Hint->ai_addr)->buf);
+    TmpHint.ai_canonname = reinterpret_cast<char *>(Hint->ai_canonname);
+    TmpHint.ai_next = nullptr;
+
+    ::getaddrinfo(Node, Service, &TmpHint, &TmpResult);
```

Comment:
`getaddrinfo` has a return value. It returns 0 if it succeeds, however, it may have several error code.[1] But it is not handled here. Please check the error code and return error message if the `getaddrinfo` is failed. 

[1]: https://man7.org/linux/man-pages/man3/getaddrinfo.3.html

Reply:
`getaddrinfo` has ten error codes, like `EAI_BADFLAGS`.  some error codes were not defined in  `enum __wasi_errno_t` .should i add some items in `__wasi_errno_t` to make `getaddrinfo` 's ten error codes has corresponding items one-to-one or deal with ten error codes in one item of '__wasi_errno_t`?

Reply:
I prefer to add new definitions in `__wasi_errno_t` to handle these error codes. These error codes are very important when debugging.